src/reactors/V2DutchOrderReactor.sol

```diff
@@ -0,0 +1,142 @@
+    error InvalidInputOverride();
+
+    /// @notice thrown when an order's output override is greater than the specified
+    error InvalidOutputOverride();
+
+    constructor(IPermit2 _permit2, address _protocolFeeOwner) BaseReactor(_permit2, _protocolFeeOwner) {}
+
+    /// @inheritdoc BaseReactor
+    function resolve(SignedOrder calldata signedOrder)
+        internal
+        view
+        virtual
+        override
+        returns (ResolvedOrder memory resolvedOrder)
+    {
+        CosignedV2DutchOrder memory cosignedOrder = abi.decode(signedOrder.order, (CosignedV2DutchOrder));
+        _validateOrder(cosignedOrder);
+        V2DutchOrder memory order = cosignedOrder.order;
+        _updateWithOverrides(order);
+
+        resolvedOrder = ResolvedOrder({
+            info: order.inner.info,
+            input: order.inner.input.decay(order.decayStartTime, order.decayEndTime),
+            outputs: order.inner.outputs.decay(order.decayStartTime, order.decayEndTime),
+            sig: signedOrder.sig,
+            hash: order.hash()
+        });
+        resolvedOrder.handleOverride(order.exclusiveFiller, order.decayStartTime, order.exclusivityOverrideBps);
+    }
+
+    /// @inheritdoc BaseReactor
+    function transferInputTokens(ResolvedOrder memory order, address to) internal override {
+        permit2.permitWitnessTransferFrom(
+            order.toPermit(),
+            order.transferDetails(to),
+            order.info.swapper,
+            order.hash,
+            V2DutchOrderLib.PERMIT2_ORDER_TYPE,
+            order.sig
+        );
+    }
+
+    function _updateWithOverrides(V2DutchOrder memory order) internal pure {
+        if (order.inputOverride != 0) order.inner.input.startAmount = order.inputOverride;
+
+        for (uint256 i = 0; i < order.inner.outputs.length; i++) {
+            DutchOutput memory output = order.inner.outputs[i];
+            uint256 outputOverride = order.outputOverrides[i];
+            if (outputOverride != 0) output.startAmount = outputOverride;
+        }
+    }
+
+    /// @notice validate the dutch order fields
+    /// - deadline must be greater than or equal than decayEndTime
+    /// - decayEndTime must be greater than or equal to decayStartTime
+    /// - if there's input decay, outputs must not decay
+    /// @dev Throws if the order is invalid
+    function _validateOrder(CosignedV2DutchOrder memory cosigned) internal pure {
+        V2DutchOrder memory order = cosigned.order;
+        if (order.inner.info.deadline < order.decayEndTime) {
+            revert DeadlineBeforeEndTime();
+        }
+
+        if (order.decayEndTime < order.decayStartTime) {
+            revert OrderEndTimeBeforeStartTime();
+        }
+
+        (bytes32 r, bytes32 s) = abi.decode(cosigned.signature, (bytes32, bytes32));
+        uint8 v = uint8(cosigned.signature[64]);
+        address signer = ecrecover(keccak256(abi.encode(order)), v, r, s);
+        if (order.inner.cosigner != signer) {
```

Comment:
i think you need to check that cosigner isnt address(0) because ecrecover returns address(0) when it errors

Reply:
good call